### PR TITLE
Fix for issue #21

### PR DESCRIPTION
**Pull request by ig0774**
Essentially, here's how I reproduced #21:

* Open a new window with a tex document and a non-editable buffer (**Settings** works nicely)
* Reload the window (**Window: Reload** from the palette)

The underlying issue is that while there is code that attempts to ensure it is only run on tex editors that have a recognised tex extension, when `refCiteComplete()` gets run it loads the active TextEditor which may or may not be the same text editor the event was triggered on.

For most TextEditors, this means we are running unnecessary code which may cause unexpected behaviour, but only with something like the **Settings** view which is a TextEditor but does not support the `getCursorPosition()` function do we get an actual error.

Long story short, this PR passes the observed TextEditor to `refCiteComplete` obviating the above and fixing #21.

In addition, since it appears that `onDidStopChanging` can be triggered on a non-active buffer, I added some guard code to ensure that the completions are only run if the buffer is the active buffer

### CODE REVIEW

**Comment by msiniscalchi** [comment]
You beat me to it :) Thanks @ig0774 !
